module/account/src/Application/Security/Voter/UserRoleVoter.php

```diff
@@ -60,16 +60,18 @@ public function voteOnAttribute($attribute, $subject, TokenInterface $token): bo
             throw new \RuntimeException(sprintf('Role by id "%s" not found', $user->getRoleId()->getValue()));
         }
 
-        $result = false;
+        $privileges = $this->query->getPrivilegesEndPointByBusiness($role->getPrivileges());
+
+
         $attributePrivilege = new Privilege($attribute);
-        /** @var Privilege $privilege */
-        foreach ($role->getPrivileges() as $privilege) {
-            if ($privilege->isEqual($attributePrivilege)) {
-                $result = true;
-                break;
+
+        foreach ($privileges as $privilege) {
+            $privilegeObject = new Privilege($privilege);
```

Comment:
$privilegeObject it not good name use $privilege or $newPrivilege or something.
You can also return privilege object collection from query then this code will be clener